### PR TITLE
ENH: New config system

### PR DESCRIPTION
**Pull request by danielballan**
As outlined in #194. For now we're targeting the current Broker structure (1 metadatstore + 1 assets). In the future we'll reorganize into (1 header source + N event sources + N assets).

### CODE REVIEW

File: databroker/broker.py [hunk 1]
```diff
@@ -170,6 +172,56 @@ def __iter__(self):
                         break
 
 
+CONFIG_SEARCH_PATH = (os.path.expanduser('~/.config/databroker/'),
```
Comment (tacaswell) [inline databroker/broker.py]: can we also include the conda env etc?

File: databroker/tests/test_config.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        'class': 'MDSRO',
+        'config': {
+            'host': 'localhost',
+            'port': 27017,
+            'database': 'mds_database_placholder',
+            'timezone': 'US/Eastern'}
+    },
+    'assets': {
+        'module': 'databroker.assets.mongo',
+        'class': 'FileStoreRO',
```
Comment (tacaswell) [inline databroker/tests/test_config.py]: This needs to be `RegistryRO` I think.

File: databroker/broker.py (hunk 1 above)
Comment (danielballan) [reply databroker/broker.py]: Do we want to special-case conda here? We did that previously, but in retrospect it seems like kind of a hacked thing to do. Is there any reason we need it, given our plans for the config deployment?

Comment (tacaswell) [reply databroker/broker.py]: I think a clean way to do this is :

```python
import sys
from pathlib import Path
local_etc = Path(sys.executable).parents[1] / 'etc' / 'databroker'
```

which will work without activation and with venvs.

Comment (danielballan) [reply databroker/broker.py]: Is there a py2-friendly equivalent of `Path(sys.executable).parents[1]`?

Comment (danielballan) [reply databroker/broker.py]: Oh, I see what that does. Easy enough.

**Comment by tacaswell** [comment]
:+1: Can you add a test of the legacy stuff?

**Pull request merged**